server/api/common/common.go

```diff
@@ -169,23 +182,42 @@ func (tm TargetMap) ProcessEvent(iid string, event *bespb.BuildEvent) {
 				Language: language,
 				Tag:      p.Configured.GetTag(),
 			}
+			if tm.selector != nil && TargetMatchesSelector(target, tm.selector) {
+				tm.M[label] = target
+			}
 		}
 	case *bespb.BuildEvent_Completed:
 		{
-			if target := tm[event.GetId().GetTargetCompleted().GetLabel()]; target != nil {
+			if target := tm.M[event.GetId().GetTargetCompleted().GetLabel()]; target != nil {
 				target.Status = cmnpb.Status_BUILT
 			}
 		}
 	case *bespb.BuildEvent_TestSummary:
 		{
-			if target := tm[event.GetId().GetTestSummary().GetLabel()]; target != nil {
+			if target := tm.M[event.GetId().GetTestSummary().GetLabel()]; target != nil {
 				target.Status = TestStatusToStatus(p.TestSummary.GetOverallStatus())
 				target.Timing = TestTimingFromSummary(p.TestSummary)
 			}
 		}
 	}
 }
 
+func TargetMatchesSelector(target *apipb.Target, selector *apipb.TargetSelector) bool {
```

Comment:
Can you add a comment on this function? (It's exported, and part of a library)

Reply:
Added comment doc